static/javascript.js

```diff
@@ -31,6 +31,11 @@
         .hide();
     };
     message.appendChild(button);
+    // Remove any previously existing messages(success and error).
+    var existingMessage = document.querySelector('.message-display').querySelector('.notice');
+    if (existingMessage !== null) {
+      document.querySelector('.message-display').querySelector('.notice').remove();
+    } 
```

Comment:
A good start, but let's enhance this a bit:

```suggestion
    var existingMessages = document.querySelectorAll('.message-display > .notice');
    Array.prototype.forEach.call(existingMessages, function (msg) {
      msg.remove();
    });
```

A couple of notes:
- it is always a good idea to ensure all message elements have been removed from the list. Your approach only removed the first element, and it will break if multiple messages should ever appear to the list through the change we make to the plugin;
- `querySelector` and `querySelectorAll` support CSS selectors, which means you can use CSS rules to match only specific elements. Your approach would match every element inside `.message-display` element, even if it is nested within another `.notice` element. It would be more efficient to only target first-level elements inside `.message-display` element. And that can be achieved with CSS rule `a > b` where `b` needs to be the immediate child to `a`. And the added benefit of this approach is that you are traversing DOM tree only once (instead of 4 times);
- the beauty of using `querySelectorAll` is that you always get a `NodeList` object as a result of the callback. This way you do not need to worry if element was found or not, and you can simply iterate over the object always;